src/Lean/Elab/PreDefinition/MkInhabitant.lean

```diff
@@ -47,6 +48,13 @@ def mkInhabitantFor (declName : Name) (xs : Array Expr) (type : Expr) : MetaM Ex
   | some val => return val
   | none     => match (← go? true) with
     | some val => return val
-    | none     => throwError "failed to compile partial definition '{declName}', failed to show that type is inhabited and non empty"
+    | none     => throwError "\
+      failed to compile partial definition '{declName}', could not prove that type is nonempty\n\
+      \n\
+      Possible solutions:\n\
+      - Ensure that there is an '{MessageData.ofConstName ``Inhabited}' or '{MessageData.ofConstName ``Nonempty}' \
+      instance for{MessageData.nest 4 <| Format.line ++ type}\n\
```

Comment:
```suggestion
      instance for '{MessageData.nest 4 <| Format.line ++ type}'\n\
```
maybe?  This is untested completely, since I have no idea of how to potentially test it!  :smile:

Reply:
I haven't tested it, but I'm pretty sure that would make the error message come out as
```
failed to compile partial definition 'checkMyList', could not prove that type is nonempty

Possible solutions:
- Ensure that there is an 'Inhabited' or 'Nonempty' instance for '
    Bool × ListNode'
- Add such an instance to the parameter list.
- Add a parameter of this type to the parameter list.
```
It's common for expressions to be printed indented without any delimiters, so I'm inclined not to add `'`'s

Reply:
Ok, I agree with you on the output, given the test, actually.  I also had not noticed that single quotes were not used in this context.